### PR TITLE
Deconstructing the wall the captain's spare safe is hanging on no longer opens the safe for free

### LINKED ISSUE #80588: deconstructing the wall the spare id safe is on drops the spare
Reporting client version: 515.1623

## Issue Summary

destroying the wall under the spare safe
destroys the safe and drops the spare

## Round ID:
[221164](https://scrubby.melonmesa.com/round/221164)

## Testmerges:
- [Save json savefiles to disk in pretty print form](https://www.github.com/tgstation/tgstation/pull/80512)
- [(Testmerge Ready) New Antagonist: Tiger Cooperative Fanatic](https://www.github.com/tgstation/tgstation/pull/80313)
- [[TM ONLY] Remove 'natural' beheading](https://www.github.com/tgstation/tgstation/pull/80439)

## Reproduction:

1. destroy wall supporting safe
2. profit

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

See title.
## Why It's Good For The Game

Haha, oops!
Fixes #80588 

## Changelog
:cl:
fix: Destroying the wall holding up the spare ID safe no longer opens said safe.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Not a workable fix. The safe cannot hang in the air, partially because it reintroduces #22283, but also because of #https://github.com/wall-nerds/wallening/issues/90

**Comment by SlendyOfDaMawn** [comment]
Perhaps introducing a "damaged safe" item?

Can't be opened via pin but can be opened with tools and a lot of patience?

**Comment by SomeoneYouProbablyKnow** [comment]
> Can't be opened via pin but can be opened with tools and a lot of patience?

Finally an excuse to add a big construction circular saw. But really a chainsaw could be a neat requirement for this.

**Comment by SlendyOfDaMawn** [comment]
> > Can't be opened via pin but can be opened with tools and a lot of patience?
> 
> Finally an excuse to add a big construction circular saw. But really a chainsaw could be a neat requirement for this.

Yeah, should probably have some long process like:

- unscrew
- wirecut
- unweld
- crowbar
- chainsaw for like 30 fucking seconds
- wirecut
- crowbar

File: code/game/objects/structures/secure_safe.dm [hunk 1]
```diff
@@ -11,16 +11,23 @@
...
 	//this will create the storage for us.
 	AddComponent(/datum/component/lockable_storage)
-	find_and_hang_on_wall()
+	find_and_hang_on_wall(custom_drop_callback = CALLBACK(src, PROC_REF(handle_wall_deconstruct)))
 	PopulateContents()
 
+/obj/structure/secure_safe/proc/handle_wall_deconstruct()
+	if(!deconstruct_on_wall_deconstruct)
+		return
+	deconstruct(disassembled = FALSE)
```
Comment (SyncIt21) [inline code/game/objects/structures/secure_safe.dm:29]:
```suggestion
	set_anchored(FALSE)
```

It now becomes an item you can drag around (but can't pick up which is intended) See


https://github.com/tgstation/tgstation/assets/110812394/bf6ec69a-0e12-4ff5-a38b-b7195f8af891

**Comment by SlendyOfDaMawn** [comment]
Or do that.
That's cool too.

**Comment by LemonInTheDark** [comment]
In tears

**Comment by SlendyOfDaMawn** [comment]
😭

**Pull request closed without merging**